### PR TITLE
[DRAFT] librewolf: init at 90.0.2

### PR DESCRIPTION
**Pull request by onny**
###### Motivation for this change
Packaging Librewolf, a privacy oriented Firefox fork, as requested in this issue https://github.com/NixOS/nixpkgs/issues/94918 .

I created a new package since I'm unsure about integrating this as a Firefox sub-package. There are no existing Firefox forks yet, just ESR, developer and beta builds. Also the Firefox packages seems to complicated for me. Maybe we could move duplicate code into a general Gecko-browsers meta package with a future rework.

This PR is a draft. I copied the Palemoon package and trying to integrate Librewolf patches + Firefox config into this.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/browsers/librewolf/default.nix [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+}:
+
+# Only specific GCC versions are supported with branding
+# https://developer.palemoon.org/build/linux/
+assert stdenv.cc.isGNU;
+assert with lib.strings; (
+  versionAtLeast stdenv.cc.version "4.9"
+  && !hasPrefix "6" stdenv.cc.version
+  && versionOlder stdenv.cc.version "11"
+);
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/browsers/librewolf/default.nix:44]: Please use meta.broken for this kind of stuff.

File: pkgs/applications/networking/browsers/librewolf/default.nix [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+
+let
+  libPath = lib.makeLibraryPath [
+    ffmpeg
+    libpulseaudio
+  ];
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/browsers/librewolf/default.nix:50]:
```suggestion
```

File: pkgs/applications/networking/browsers/librewolf/default.nix [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+    rm -rf $out/{include,share/idl,lib/palemoon-devel-${version}}
+
+    runHook postInstall
+  '';
+
+  dontWrapGApps = true;
+
+  preFixup = ''
+    gappsWrapperArgs+=(
+      --prefix LD_LIBRARY_PATH : "${libPath}"
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/browsers/librewolf/default.nix:216]:
```suggestion
      --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath [ ffmpeg libpulseaudio ]}"
```

File: pkgs/applications/networking/browsers/librewolf/default.nix [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+    # Install official branding icons
+    for iconname in default{16,22,24,32,48,256} mozicon128; do
+      n=''${iconname//[^0-9]/}
+      size=$n"x"$n
+      install -Dm644 ./palemoon/branding/official/$iconname.png $out/share/icons/hicolor/$size/apps/palemoon.png
+    done
+
+    # Remove unneeded SDK data from installation
+    # https://forum.palemoon.org/viewtopic.php?f=37&t=26796&p=214676#p214729
+    rm -rf $out/{include,share/idl,lib/palemoon-devel-${version}}
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/browsers/librewolf/default.nix:207]:
```suggestion
    rm -r $out/{include,share/idl,lib/palemoon-devel-${version}}
```

File: pkgs/applications/networking/browsers/librewolf/default.nix [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+    gtk2
+    libevent
+    libGL
+    libGLU
+    libnotify
+    libpulseaudio
+    libstartup_notification
+    zlib
+  ]
+  ++ (with xorg; [
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/browsers/librewolf/default.nix:109]:
```suggestion
  ] ++ (with xorg; [
```

File: pkgs/applications/networking/browsers/librewolf/default.nix [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+    update-source-version ${pname} "$version"
+  '';
+
+  nativeBuildInputs = [
+    autoconf213
+    desktop-file-utils
+    gnum4
+    perl
+    pkg-config
+    python2
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/browsers/librewolf/default.nix:83]: Do we really need to use python2?

**Comment by etircopyh** [comment]
I'd personally rather use firedragon. It seems to have more reasonable defaults. Surprised it hasn't been added to nixpkgs yet.

**Comment by cheriimoya** [comment]
i think this is done in #151439

**Pull request closed without merging**